// district-data/dist/cjs/source/baseSource.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/source/baseSource.ts
var baseSource_exports = {};
__export(baseSource_exports, {
  default: () => BaseSource
});
module.exports = __toCommonJS(baseSource_exports);
var BaseSource = class {
  constructor(options) {
    this.info = {};
    this.options = {};
    this.data = {
      country: void 0,
      province: void 0,
      city: void 0,
      county: void 0,
      jiuduanxian: void 0
    };
    this.fetchJsonData = async (url) => {
      const res = await fetch(url);
      return await res.json();
    };
    this.options = {
      ...this.getDefaultOptions(),
      ...options
    };
    this.version = this.options.version || "default";
    this.type = this.options.type || "wgs84";
  }
};
