// district-data/dist/cjs/source/datavSource.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/source/datavSource.ts
var datavSource_exports = {};
__export(datavSource_exports, {
  DataVSource: () => DataVSource
});
module.exports = __toCommonJS(datavSource_exports);
var import_baseSource = __toESM(require("./baseSource"));
var DataConfig = {
  desc: {
    text: "DataV",
    href: "https://datav.aliyun.com/portal/school/atlas/area_selector"
  },
  url: "https://geo.datav.aliyun.com"
};
var DataVSource = class extends import_baseSource.default {
  constructor() {
    super(...arguments);
    this.info = DataConfig;
  }
  getDefaultOptions() {
    return {
      version: "areas_v3"
    };
  }
  getRenderData(options) {
    throw new Error("Method not implemented.");
  }
  getData(options) {
    const { code, full } = options;
    const data = this.fetchData(code, full);
    return data;
  }
  getChildrenData(ChildrenDataOptions) {
    const { parentName, full } = ChildrenDataOptions;
    const data = this.fetchData(parentName, full);
    return data;
  }
  async fetchData(code, full) {
    let url = full ? `${DataConfig.url}/${this.version}/bound/${code}_full.json` : `${DataConfig.url}/${this.version}/bound/${code}.json`;
    if (this.options.getCdnUrl) {
      url = this.options.getCdnUrl({
        origin: DataConfig.url,
        version: this.version,
        adcode: code,
        isFull: full
      });
    }
    return await this.fetchJsonData(url);
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DataVSource
});
