// district-data/dist/cjs/source/rdbSource.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/source/rdbSource.ts
var rdbSource_exports = {};
__export(rdbSource_exports, {
  RDBSource: () => RDBSource
});
module.exports = __toCommonJS(rdbSource_exports);
var import_simplify = __toESM(require("@turf/simplify"));
var import_geobuf = __toESM(require("geobuf"));
var import_pbf = __toESM(require("pbf"));
var import_baseSource = __toESM(require("./baseSource"));
var DataConfig = {
  desc: {
    text: "锐多宝的地理空间",
    href: "https://github.com/ruiduobao/shengshixian.com"
  },
  url: "https://cdn.cbd.int/xingzhengqu"
  // url: 'https://unpkg.com/xingzhengqu',
};
var DataAccuracy = {
  high: 1e-6,
  middle: 1e-5,
  low: 5e-3
};
var DataLevelRecord = {
  country: "country",
  province: "province",
  city: "city",
  county: "county",
  jiuduanxian: "jiuduanxian"
};
var RDBSource = class extends import_baseSource.default {
  constructor() {
    super(...arguments);
    this.info = DataConfig;
    /*  */
    this.fetchArrayBuffer = async (url) => {
      const res = await fetch(url);
      return await res.arrayBuffer();
    };
  }
  getDefaultOptions() {
    return {
      version: "2023"
    };
  }
  // 使用 Low 精度数据进行数据渲染
  async getRenderData(options) {
    return this.getData(options);
  }
  // 获取数据
  async getData(options) {
    const { level = "country" } = options;
    return await this.fetchData(level);
  }
  // 获取子级数据,数据下载时使用
  async getChildrenData(ChildrenDataOptions) {
    const {
      parentAdcode,
      parentLevel,
      childrenLevel,
      precision = "low"
    } = ChildrenDataOptions;
    const rawData = await this.getData({ level: childrenLevel, precision });
    let resultFeatures;
    if (parentAdcode && parentLevel && parentAdcode !== 1e5) {
      resultFeatures = rawData.features.filter((feature) => {
        const key = `${parentLevel}_adcode`;
        const code = feature.properties[key];
        return code === parentAdcode;
      });
    } else {
      resultFeatures = rawData.features;
    }
    return {
      type: "FeatureCollection",
      features: resultFeatures || []
    };
  }
  // 获取原始数据，数据解析并缓存
  async fetchData(level) {
    if (this.data[level]) {
      return this.data[level];
    }
    let url = +this.version >= 2024 ? `${DataConfig.url}@${this.version}/data/${this.type}/${DataLevelRecord[level]}.pbf` : `${DataConfig.url}@${this.version}/data/${DataLevelRecord[level]}.pbf`;
    if (this.options.getCdnUrl) {
      url = this.options.getCdnUrl({
        origin: DataConfig.url,
        version: this.version,
        level: DataLevelRecord[level],
        type: this.type
      });
    }
    const data = await this.fetchArrayBuffer(url);
    const jsonData = import_geobuf.default.decode(new import_pbf.default(data));
    this.data[level] = jsonData;
    return jsonData;
  }
  simplifyData(data, precision) {
    return (0, import_simplify.default)(data, {
      tolerance: DataAccuracy[precision],
      highQuality: false
    });
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  RDBSource
});
